### PR TITLE
Cherry-pick: Fix incorrect EnableTCPReset for non-TCP protocols

### PR DESCRIPTION
**Pull request by lzhecheng**
Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
This code change fixes the bug that UDP services would trigger unnecessary LoadBalancer updates. The root cause is that a field not working for non-TCP protocols is considered.
ref: https://github.com/kubernetes-sigs/cloud-provider-azure/pull/1090
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by lzhecheng** [comment]
@nilo19 Could you help trigger the test?

**Comment by lzhecheng** [comment]
<html>
<body>


/test pull-kubernetes-e2e-capz-conformance
--



</body>
</html>

**Review by feiskyer — APPROVED** [review]
/lgtm

**Comment by feiskyer** [comment]
@andrewsykim  could you help to have a look at this bug fix?

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go
Comment (andrewsykim) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go:1705]: This can just be `tcpReset = nil` and you can get rid of `nilTCPReset`?

**Review by andrewsykim — COMMENTED** [review]
minor comment about a variable that can be deleted

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go
Comment (lzhecheng) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go]: Good catch. Updated.

**Review by feiskyer — APPROVED** [review]
/lgtm

**Comment by feiskyer** [comment]
/assign @nckturner 

@nckturner could you help to have a look?

**Comment by lzhecheng** [comment]
/assign @cheftako
Could you take a look? Thanks!

**Comment by lzhecheng** [comment]
@andrewsykim could you help take another look?

**Review by andrewsykim — COMMENTED** [review]
/approve
/hold

Just ack'ing that this is bug fix, deferring final lgtm to Azure maintainrs

**Comment by lzhecheng** [comment]
@andrewsykim I think @feiskyer has reviewed the PR and lgtm. Could you please assign someone else you think should review?

**Review by feiskyer — APPROVED** [review]
/lgtm
/hold cancel

**Pull request merged**
